bibernate-core/src/main/java/com/bobocode/petros/bibernate/session/jdbc/DefaultJdbcQueryManager.java

```diff
@@ -1,46 +1,162 @@
+import lombok.extern.slf4j.Slf4j;
 
 import javax.sql.DataSource;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
+import static java.sql.PreparedStatement.RETURN_GENERATED_KEYS;
+
+/**
+ * Implementation of {@link JdbcQueryManager}.
+ */
+@Slf4j
 public class DefaultJdbcQueryManager implements JdbcQueryManager {
 
-    private DataSource dataSource;
+    private static final String GENERATED_KEY = "GENERATED_KEY";
+
+    private final DataSource dataSource;
 
-    public DefaultJdbcQueryManager(DataSource dataSource) {
-        Objects.requireNonNull(dataSource, "Parameter [dataSource] must be provided!");
+    public DefaultJdbcQueryManager(final DataSource dataSource) {
+        Objects.requireNonNull(dataSource, "Parameter [dataSource] must not be null!");
         this.dataSource = dataSource;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param entity entity that is supposed to be persisted
+     * @param <T>    generic type
+     * @return persisted entity with id
+     */
     @Override
-    public <T> T persist(T entity) {
-        return null;
+    public <T> T persist(final T entity) {
+        Objects.requireNonNull(entity, "Parameter [entity] must not be null!");
+        final String sql = QueryBuilder.buildQuery(entity.getClass(), QueryType.INSERT, Collections.emptyList());
+        try (final Connection connection = this.dataSource.getConnection();
+             final PreparedStatement statement = StatementUtils.prepareInsertStatement(connection.prepareStatement(sql, RETURN_GENERATED_KEYS), entity)) {
+            statement.executeUpdate();
+            final ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                final Field field = EntityUtils.getIdField(entity.getClass());
+                field.setAccessible(true);
+                field.set(entity, generatedKeys.getObject(GENERATED_KEY, EntityUtils.getIdField(entity.getClass()).getType()));
+            }
+            log.debug("Query: '{}', result: {}", sql, entity);
+            return entity;
+        } catch (SQLException | IllegalAccessException e) {
+            throw new JdbcOperationException(e.getMessage(), e);
+        }
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param type         class of an entity to search for
+     * @param restrictions applicable restrictions
+     * @param <T>          generic type
+     * @param <C>          collection of type T
+     * @return collection of entities
+     */
     @Override
-    public <T> T find(Class<T> type, List<Restriction> restrictions) {
-        return null;
+    public <T, C extends Collection<T>> C find(final Class<T> type, final List<Restriction> restrictions) {
```

Comment:
Can't find a reason for `C extends Collection<T>`. I think we can simplify this to
`<T> Collection<T> find(final Class<T> type, final List<Restriction> restrictions)`